### ISSUE TITLE
macOS packaged release unable to flashfwpkg

### ISSUE THREAD

**Issue by apavely** [HewlettPackard/python-redfish-utility#41]
Since release v3.1.0 of the packaged binaries for macOS, I have been unable to run `flashfwpkg` due to a missing "iLOrest Chif dll". 
```
$ ilorest-3.2.2 flashfwpkg U34_2.52_07_08_2021.fwpkg --url testhost-ilo5.domain.local -u admin -p 'XXXXXXXXXXXX'
iLOrest : RESTful Interface Tool version 3.2.2
Copyright (c) 2014-2021 Hewlett Packard Enterprise Development LP
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Discovering data...Done
iLOrest Chif dll not found, please check that the chif dll is present.
```
```
$ ilorest-3.1.1 flashfwpkg U34_2.52_07_08_2021.fwpkg --url testhost-ilo5.domain.local -u admin -p 'XXXXXXXXXXXX'
iLOrest : RESTful Interface Tool version 3.1.1
Copyright (c) 2014-2020 Hewlett Packard Enterprise Development LP
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Discovering data...Done
WARNING: Cache is activated. Session keys are stored in plaintext.
iLOrest Chif dll not found, please check that the chif dll is present.
```
```
$ ilorest-3.1.0 flashfwpkg U34_2.52_07_08_2021.fwpkg --url testhost-ilo5.domain.local -u admin -p 'XXXXXXXXXXXX'
iLOrest : RESTful Interface Tool version 3.1
Copyright (c) 2014-2020 Hewlett Packard Enterprise Development LP
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Discovering data...Done
WARNING: Cache is activated. Session keys are stored in plaintext.
iLOrest Chif dll not found, please check that the chif dll is present.
```
It last ran successfully using v3.0.1.
```
$ ilorest-3.0.1 flashfwpkg U34_2.52_07_08_2021.fwpkg --url testhost-ilo5.domain.local -u admin -p 'XXXXXXXXXXXX'
iLOrest : RESTful Interface Tool version 3.0.1
Copyright (c) 2014, 2019 Hewlett Packard Enterprise Development LP
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Discovering data...Done
WARNING: Cache is activated. Session keys are stored in plaintext.
Uploading firmware: U34_2.52_07_08_2021.signed.flash
[200] The operation completed successfully.
Component U34_2.52_07_08_2021.signed.flash uploaded successfully
Waiting for iLO UpdateService to finish processing the component
0 hour(s) 1 minute(s) 40 second(s)
Firmware has successfully been flashed and a reboot is required for this firmware to take effect.
```
As I understand it, CHIF is the local iLO channel interface, and accessible only locally, so is there a reason we're triggering the CHIF code path for remote (HTTP) connections?

**Comment by rajeevkallur** [comment]
Does it work if IP address used instead of system name?  Tested with latest code, it seems to work fine.
![image](https://user-images.githubusercontent.com/21117913/130437737-50100a35-2f1d-4671-9499-e1b595842e41.png)

**Comment by rajeevkallur** [comment]
Please let us know if it is still an issue.

**Comment by apavely** [comment]
Still an issue.

It makes no difference whether I specify an IP address or a DNS hostname, or whether or not I include the full URI prefix. I have tried multiple iterations without success, with and without various flags, even mimicking yours above:

```
$ ilorest-3.2.2 -v flashfwpkg U34_2.52_07_08_2021.fwpkg --forceupload --tpmover --url 172.16.0.10 -u admin -p 'XXXXXXXXXXXX'
iLOrest : RESTful Interface Tool version 3.2.2
Copyright (c) 2014-2021 Hewlett Packard Enterprise Development LP
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Discovering data...Done
iLOrest Chif dll not found, please check that the chif dll is present.
ILOREST return code: 35
```

Just to be clear, it's not just the released binaries for macOS ... it's also present with the latest HEAD:
```
$ which python3
/usr/local/bin/python3
$ python3 -V
Python 3.9.6
$ python3 rdmc.py -v flashfwpkg ~/Downloads/fwpkg/U34_2.52_07_08_2021.fwpkg --forceupload --tpmover --url 172.16.0.10 -u admin -p 'XXXXXXXXXXXX'
iLOrest : RESTful Interface Tool version 3.2.2
Copyright (c) 2014-2021 Hewlett Packard Enterprise Development LP
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Discovering data...Done
iLOrest Chif dll not found, please check that the chif dll is present.
ILOREST return code: 35
```

This is a problem that appears to be specific to the macOS versions released due to the lack of a CHIF library for the macOS platform (I am running macOS 10.15.7 Catalina on Intel). I can understand why there's no CHIF library available for macOS, as no HPE hardware is designed to run macOS. As a remote client platform, though, I regularly use macOS to script firmware updates to our iLO5-based HPE Gen10 servers.

In other words, if we're using Redfish over HTTPS for connectivity, why are we breaking on a CHIF requirement? It shouldn't matter what platform you are on. Are there any code paths that explicitly depend on CHIF being present for remote  connections (Redfish over HTTPS)?

**Comment by rajeevkallur** [comment]
Does this sequence work?  (explicit logging initially)
python3 rdmc.py login 172.16.0.10 -u admin -p 'XXXXXXXXXXXX'
python3 rdmc.py -v flashfwpkg ~/Downloads/fwpkg/U34_2.52_07_08_2021.fwpkg --forceupload --tpmover
